### PR TITLE
Unit tests for AxisAlignedBoundingBox

### PR DESCRIPTION
**Pull request by mattj23**
Added 38 unit tests to capture the existing behavior of the `AxisAlignedBoundingBox`, including the particular behavior around the `+=` operator.  

The `AxisAlignedBoundingBox` has been in Open3D since 2019, and at this point changes to its behavior would likely break user code. These tests should provide a warning that will catch unintentional behavior changes.

---
This change is [Reviewable](https://reviewable.io/reviews/intel-isl/open3d/3433)

### CODE REVIEW

File: cpp/tests/geometry/AxisAlignedBoundingBox.cpp
Comment (reyanshsolis) [inline cpp/tests/geometry/AxisAlignedBoundingBox.cpp:33]: We may remove this comment, replace `v_t` back to `Vector3d` (we may use `using Eigen::Vector3d`). 

For formatting:
You may run `make apply-style` from inside your build folder if you are using Mac or Ubuntu. [More about automated style formatting: [Here](http://www.open3d.org/docs/release/contribute/styleguide.html)].

File: cpp/tests/geometry/AxisAlignedBoundingBox.cpp
Comment (reyanshsolis) [inline cpp/tests/geometry/AxisAlignedBoundingBox.cpp:43]: We may also remove this comment, as it might be obvious for a unit test.

File: cpp/tests/geometry/AxisAlignedBoundingBox.cpp
Comment (reyanshsolis) [inline cpp/tests/geometry/AxisAlignedBoundingBox.cpp:299]: Please add an extra line at the end of the file.

**Review by reyanshsolis — CHANGES_REQUESTED** [review]
Thanks a lot for your valuable contribution. I have added some minor comments. Also, please merge master to this branch.

File: cpp/tests/geometry/AxisAlignedBoundingBox.cpp
Comment (mattj23) [reply cpp/tests/geometry/AxisAlignedBoundingBox.cpp]: replaced `v_t` with `Vector3d`

File: cpp/tests/geometry/AxisAlignedBoundingBox.cpp
Comment (mattj23) [reply cpp/tests/geometry/AxisAlignedBoundingBox.cpp]: Comment removed

File: cpp/tests/geometry/AxisAlignedBoundingBox.cpp
Comment (mattj23) [reply cpp/tests/geometry/AxisAlignedBoundingBox.cpp]: newline added

**Review by mattj23 — COMMENTED** [review]
Thank you for the review.  I made the requested changes, although one of the CI builds seems to have failed with a broken dependency that I believe isn't related to my PR. Do I need to update the CHANGELOG.md as the bot requested?

> *[Reviewable](https://reviewable.io/reviews/intel-isl/open3d/3433#-:-M_wYZTfC4kH6c1CFk8r:b-dm7rcr)* status: 0 of 2 files reviewed, 3 unresolved discussions (waiting on @reyanshsolis and @theNded)

**Review by reyanshsolis — APPROVED** [review]
Thanks, @mattj23. The PR looks good to me. Please wait for a secondary approval from @errissa or @theNded.

**Review by theNded — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/intel-isl/open3d/3433)* status: all files reviewed, 3 unresolved discussions (waiting on @errissa and @reyanshsolis)

**Pull request merged**
